mirrord-layer/src/lib.rs

```diff
@@ -80,89 +79,106 @@ fn init() {
     RUNTIME.block_on(start_layer_thread(port_forwarder, receiver, config));
 }
 
-async fn handle_hook_message(
-    hook_message: HookMessage,
-    tcp_mirror_handler: &mut TcpMirrorHandler,
-    codec: &mut actix_codec::Framed<impl AsyncRead + AsyncWrite + Unpin + Send, ClientCodec>,
-    getaddrinfo_handler: &Mutex<Vec<ResponseChannel<Vec<AddrInfoInternal>>>>,
-    file_handler: &mut FileHandler,
-) {
-    match hook_message {
-        HookMessage::Tcp(message) => {
-            tcp_mirror_handler
-                .handle_hook_message(message, codec)
-                .await
-                .unwrap();
-        }
-        HookMessage::File(message) => {
-            file_handler
-                .handle_hook_message(message, codec)
-                .await
-                .unwrap();
-        }
-        HookMessage::GetAddrInfoHook(GetAddrInfoHook {
-            node,
-            service,
-            hints,
-            hook_channel_tx,
-        }) => {
-            trace!("HookMessage::GetAddrInfo");
+struct Layer<T>
+where
+    T: tokio::io::AsyncRead + tokio::io::AsyncWrite + Unpin + Send,
+{
+    pub codec: actix_codec::Framed<T, ClientCodec>,
+    ping: bool,
+    tcp_mirror_handler: TcpMirrorHandler,
+    file_handler: FileHandler,
+    getaddrinfo_handler_queue: VecDeque<ResponseChannel<Vec<AddrInfoInternal>>>,
+}
 
-            getaddrinfo_handler.lock().unwrap().push(hook_channel_tx);
+impl<T> Layer<T>
+where
+    T: tokio::io::AsyncRead + tokio::io::AsyncWrite + Unpin + Send,
```

Comment:
Do we need this `T` generic? Don't we create only 1 type of `Layer<T>` (if you can see us needing a different `T` in the future, it's ok)?

Reply:
Forgot about the pain around the `Frame(whatever)` type.